### PR TITLE
aux: don't panic if the .aux file is not writable

### PR DESCRIPTION
**Pull request by gares**
~I'm currently validating this patch on the platform CI, hence the draft.~ it works

This PR is pushing upstream fix https://github.com/coq/platform/pull/127/commits/2f55ee7ea0d354d4a8cc53f58fff2d01a2b06d97 from pr https://github.com/coq/platform/pull/127

### CODE REVIEW

File: lib/aux_file.ml [hunk 1]
```diff
@@ -30,12 +30,14 @@ let mk_absolute vfile =
...
 let start_aux_file ~aux_file:output_file ~v_file =
   let vfile = mk_absolute v_file in
-  oc := Some (open_out output_file);
-  Printf.fprintf (Option.get !oc) "COQAUX%d %s %s\n"
-    version (Digest.to_hex (Digest.file vfile)) vfile
+  try
+    oc := Some (open_out output_file);
+    Printf.fprintf (Option.get !oc) "COQAUX%d %s %s\n"
+      version (Digest.to_hex (Digest.file vfile)) vfile
+  with Sys_error _ -> ()
```
Comment (Alizter) [inline lib/aux_file.ml:37]: Wouldn't it be better to print a warning instead of doing nothing?

Comment (gares) [reply lib/aux_file.ml:37]: In general I'd say yes, but in this particular case I believe it is best this way for two reasons:
- if the user is really using a snap package, he can't do much to fix the warning (other than granting the snap full power, which is defeating the whole containment idea)
- if the file system is read only, for example, then the user is going to get a good error message (I hope) from the code trying to write the .vo file

**Comment by gares** [comment]
CC @silene

**Comment by ejgallego** [comment]
Indeed I don't follow the rationale of this patch, if the .aux file can't be written, it is better to stop and exit than to run a partly broken setup IMHO.

**Comment by gares** [comment]
well, .aux files are optional. If they are not found coqc just works fine.

**Comment by ejgallego** [comment]
> well, .aux files are optional. If they are not found coqc just works fine.

This is about writing the file, not reading it right?

What is the case here where the .aux file cannot be written but the .vo can?

**Comment by gares** [comment]
> The home interface allows access to non-hidden files owned by the user in the user’s home ($HOME) directory where a user normally stores their personal files and documents.

From here https://snapcraft.io/docs/home-interface
In order to read/write dot files you need to run the snap in classic mode (no confinement) which is not nice.

**Comment by ejgallego** [comment]
> From here https://snapcraft.io/docs/home-interface
> In order to read/write dot files you need to run the snap in classic mode (no confinement) which is not nice.

Ouch that looks like PITA; does that apply to subdirectories too?

IMHO we should at least emit a warning instead of silently eating the problem. Renaming .aux files to remove the dot could be a better fix for 8.15

**Comment by gares** [comment]
At the cost of repeating myself:

In general I'd say yes, but in this particular case I believe it is best this way for two reasons:

- if the user is really using a snap package, he can't do much to fix the warning (other than granting the snap full power, which is defeating the whole containment idea)
-  if the file system is read only, for example, then the user is going to get a good error message (I hope) from the code trying to write the .vo file

**Comment by ejgallego** [comment]
> * if the user is really using a snap package, he can't do much to fix the warning (other than granting the snap full power, which is defeating the whole containment idea)

But that restriction only happens if coqc tries to write to `$HOME` right? Are subdirs OK?

**Comment by gares** [comment]
No, they are the same according to my tests.

There are options you can use to grant extra permissions to a snap, but I don't want them to be mandatory, otherwise it defeats the purpose of making it easy for newcomers.

> IMHO we should at least emit a warning instead of silently eating the problem. Renaming .aux files to remove the dot could be a better fix for 8.15

I'd like to have this stuff be put inside the .vo file as a segment (or just drop the feature). PMP already did some work in this direction.

**Comment by ejgallego** [comment]
> No, they are the same according to my tests.

In this case let's go that way.

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@coqbot: merge now

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**